acceptance/tests/language/exported_resources.rb

```diff
@@ -0,0 +1,173 @@
+test_name "C94788: exported resources using a yaml terminus for storeconfigs" do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment(app_type)
+  exported_username = 'er0ck'
+
+  teardown do
+    step 'stop puppet server' do
+      on(master, "service #{master['puppetservice']} stop")
+    end
+    step 'remove cached agent pson catalogs from the master' do
+      on(master, "rm -f #{File.join(master.puppet['yamldir'],'catalog','*')}",
+         :accept_all_exit_codes => true)
+    end
+    on(master, "mv #{File.join('','tmp','puppet.conf')} #{master.puppet['confdir']}",
+       :accept_all_exit_codes => true)
+    step 'remove the tmp environment symlink' do
+      on master, "rm -rf #{File.join(environmentpath, tmp_environment)}"
+    end
+    step 'clean out collected resources' do
+      on(hosts, puppet_resource("user #{exported_username} ensure=absent"), :accept_all_exit_codes => true)
+    end
+  end
+
+  step 'create a yaml storeconfigs terminus in the modulepath' do
+    moduledir = File.join(environmentpath,tmp_environment,'modules')
+    manifest = <<MANIFEST
+File {
+  ensure => directory,
+}
+file {
+  '#{moduledir}':;
+  '#{moduledir}/yaml_terminus':;
+  '#{moduledir}/yaml_terminus/lib':;
+  '#{moduledir}/yaml_terminus/lib/puppet':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/catalog':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/facts':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/node':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/resource':;
+}
+file { '#{moduledir}/yaml_terminus/lib/puppet/indirector/catalog/yaml_storeconfigs.rb':
+  ensure => file,
+  content => '
+    require "puppet/indirector/catalog/yaml"
+    class Puppet::Resource::Catalog::YamlStoreconfigs < Puppet::Resource::Catalog::Yaml
+      def save(request)
+        raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
+        file = path(request.key)
+        basedir = File.dirname(file)
+        # This is quite likely a bad idea, since we are not managing ownership or modes.
+        Dir.mkdir(basedir) unless Puppet::FileSystem.exist?(basedir)
+        begin
+          #Puppet::Util::Yaml.dump(request.instance, file)
+          # yeah, this is the yaml terminus, but we cannot dump anonymous modules
+          #   and we have some in the catalog (request object)
+          File.open(file, "w") { |f| f.write request.instance.to_pson }
```

Comment:
well, i could rename it pson_storeconfigs, and use that as its setting. but i'm not sure how to indicate that it's SaveOnly as the 4 termini for catalog, facts, node, resource indirection are all used based on that setting.  and the resource terminus, for instance, is certainly not SaveOnly.  
the test creates this terminus then removes it from the environment/modulepath, so it's relatively safe.  i suppose i could raise on instances of attempts to read via the catalog terminus?  but i'm not really sure how to do that offhand.
i could look into it, but it probably won't be done for this sprint, give my other work.